golang/thrift/context.go

```diff
@@ -70,6 +70,15 @@ func NewContext(timeout time.Duration) (Context, context.CancelFunc) {
 	}, cancel
 }
 
+// WrapContextForTest returns a Context that is associated with the call.
+// This should be used in units test only.
+func WrapContextForTest(ctx context.Context, call tchannel.IncomingCall) Context {
```

Comment:
Really good point. My previous approach felt a bit weird as I had 'wrap...test' at two different places which were not that different...

Fixed.